plugins/kubernetes/test/models/kubernetes/deploy_yaml_test.rb

```diff
@@ -131,6 +131,38 @@
       end
     end
 
+    describe "secret-sidecar-containers" do
+      let(:container) { result.fetch(:spec).fetch(:template).fetch(:spec).fetch(:containers).first }
+      before do
+        ENV["VAULT_ADDR"]="somehostontheinternet"
+        ENV["SECRET_SIDECAR_IMAGE"]="docker-registry.zende.sk/truth_service:secrets_sidecar_v36"
+        ENV["VAULT_SSL_VERIFY"]="false"
+      end
+
+      it "creates a sidecar" do
+        yaml.to_hash[:spec][:template][:spec][:containers].last[:name].must_equal('secret-sidecar')
+      end
+
+      it "fails if there is no primary container" do
+        assert doc.raw_template.sub!("      containers:\n      - {}", '')
+        e = assert_raises Samson::Hooks::UserError do
+          yaml.to_hash
+        end
+        e.message.must_include "has 0 containers, having 1 section is valid"
+      end
+
+      it "adds to existing volume definitions" do
+        doc.raw_template.gsub!("containers:\n      - {}\n", "containers:\n      - {}\n      volumes:\n      - {}\n      - {}\n")
+        yaml.to_hash[:spec][:template][:spec][:volumes].count.must_be(:>=, 2)
+      end
+
+      after do
```

Comment:
prefer to keep before and after together so it's easy to check symmetry ... even better to use around + with_env

```Ruby
around { |t| with_env(''VAULT" => 'bla', "BAR" => 'foo', &test) }
```